pkg/bgpv1/agent/controller.go

```diff
@@ -300,26 +300,38 @@ func PolicySelection(ctx context.Context, labels map[string]string, policies []*
 	// one policy applies to a node, we disconnect from all BGP peers and log
 	// an error.
 	for _, policy := range policies {
-		nodeSelector, err := slimmetav1.LabelSelectorAsSelector(policy.Spec.NodeSelector)
-		if err != nil {
-			l.WithError(err).Error("Failed to convert CiliumBGPPeeringPolicy's NodeSelector to a label.Selector interface")
-			continue
-		}
+		var selected bool
+
 		l.WithFields(logrus.Fields{
-			"policyNodeSelector": nodeSelector.String(),
+			"policyName":         policy.Name,
 			"nodeLabels":         slimLabels,
+			"policyNodeSelector": policy.Spec.NodeSelector.String(),
 		}).Debug("Comparing BGP policy node selector with node's labels")
-		if nodeSelector.Matches(slimLabels) {
-			if selected != nil {
+
+		if policy.Spec.NodeSelector == nil {
```

Comment:
This logic becomes a bit windy. 

Can we simply early return here, and not do the if/else later? I think this is okay, right? Since if the policy has no selected, we know we are going to select it and return it. 

```suggestion
                if policy.Spec.NodeSelector == nil {
                    return policy, nil
                }
```
Then you can get rid of the else block, and do the NodeSelector processing like normal.

Reply:
It is inconsistent with an existing behavior I think. An existing behavior is when multiple policies select the same node, `PolicySelection` returns an error and the controller resets the existing BGP session. Returning earlier here breaks this behavior.

Reply:
Yup, I see, mis understood the initial intent here :)